lib/Service/Appointments/AvailabilityGenerator.php

```diff
@@ -123,15 +122,17 @@ public function generate(AppointmentConfig $config,
 	private function filterDates(int $start, array $availabilityArray, string $timeZone) : array {
 		// First, transform all timestamps to DateTime Objects
 		$availabilityRules = [];
-		foreach($availabilityArray as $key => $availabilitySlot) {
-			if(empty($availabilitySlot)) {
+		foreach($availabilityArray as $key => $availabilitySlots) {
+			if(empty($availabilitySlots)) {
 				$availabilityRules[$key] = [];
 				continue;
 			}
-			$availabilityRules[$key] = [
-				'start' => (new DateTimeImmutable())->setTimestamp($availabilitySlot['start']),
-				'end' => (new DateTimeImmutable())->setTimestamp($availabilitySlot['end'])
-			];
+			foreach($availabilitySlots as $slot) {
+				$availabilityRules[$key][] = [
```

Comment:
still not a fan of using uninitialized array indices but since php allows this :+1: